From 559edf59921f9f3f56ac347b3d5582910c945af7 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Wed, 29 May 2019 09:12:03 +0800
Subject: [PATCH] samba: fix avc denials for smbd_t and nmbd_t

Fix:
type=AVC msg=audit(1559087800.342:77): avc:  denied  { map } for
pid=335 comm="smbd" path="/var/lib/samba/names.tdb" dev="vda" ino=31321
scontext=system_u:system_r:smbd_t:s0-s15:c0.c1023
tcontext=system_u:object_r:samba_var_t:s0 tclass=file permissive=1

type=AVC msg=audit(1559087799.797:55): avc:  denied  { map } for
pid=248 comm="nmbd" path="/var/lib/samba/names.tdb" dev="vda" ino=31321
scontext=system_u:system_r:nmbd_t:s0-s15:c0.c1023
tcontext=system_u:object_r:samba_var_t:s0 tclass=file permissive=1

Upstream-Status: Pending

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/services/samba.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/services/samba.te b/policy/modules/services/samba.te
index 305a154..3d4bda4 100644
--- a/policy/modules/services/samba.te
+++ b/policy/modules/services/samba.te
@@ -305,6 +305,7 @@ manage_dirs_pattern(smbd_t, samba_var_t, samba_var_t)
 manage_files_pattern(smbd_t, samba_var_t, samba_var_t)
 manage_lnk_files_pattern(smbd_t, samba_var_t, samba_var_t)
 manage_sock_files_pattern(smbd_t, samba_var_t, samba_var_t)
+mmap_read_files_pattern(smbd_t, samba_var_t, samba_var_t)
 files_var_filetrans(smbd_t, samba_var_t, dir, "samba")
 
 manage_dirs_pattern(smbd_t, smbd_tmp_t, smbd_tmp_t)
@@ -553,6 +554,7 @@ manage_files_pattern(nmbd_t, samba_var_t, samba_var_t)
 manage_files_pattern(nmbd_t, samba_var_t, samba_var_t)
 manage_lnk_files_pattern(nmbd_t, samba_var_t, samba_var_t)
 manage_sock_files_pattern(nmbd_t, samba_var_t, samba_var_t)
+mmap_read_files_pattern(nmbd_t, samba_var_t, samba_var_t)
 files_var_filetrans(nmbd_t, samba_var_t, dir, "nmbd")
 files_var_filetrans(nmbd_t, samba_var_t, dir, "samba")
 
-- 
2.7.4

